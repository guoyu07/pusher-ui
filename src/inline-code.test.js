import React from 'react';
import renderer from 'react-test-renderer';

import InlineCode from './inline-code';

describe('InlineCode', () => {
  const defaultProps = { children: 'const foo = "bar";' };
  const createSnapshot = (props = defaultProps) => {
    expect(
      renderer.create(<InlineCode {...props} />).toJSON(),
    ).toMatchSnapshot();
  };

  it('should render an inline codeblock element', () => {
    createSnapshot();
  });

  it('should render children', () => {
    createSnapshot({ children: <span>foo</span> });
  });

  it('should take any css props as usual', () => {
    createSnapshot({
      ...defaultProps,
      backgroundColor: 'hotpink',
      padding: 10,
    });
  });
});
