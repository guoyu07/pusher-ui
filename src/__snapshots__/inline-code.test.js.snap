// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineCode should render an inline codeblock element 1`] = `
.glamor-0,
[data-glamor-0] {
  font-size: 0.875rem;
  padding-top: 0.0625rem;
  padding-right: 0.1875rem;
  padding-bottom: 0.1875rem;
  padding-left: 0.1875rem;
}

<code
  className="glamor-0"
>
  const foo = "bar";
</code>
`;

exports[`InlineCode should render children 1`] = `
.glamor-0,
[data-glamor-0] {
  font-size: 0.875rem;
  padding-top: 0.0625rem;
  padding-right: 0.1875rem;
  padding-bottom: 0.1875rem;
  padding-left: 0.1875rem;
}

<code
  className="glamor-0"
>
  <span>
    foo
  </span>
</code>
`;

exports[`InlineCode should take any css props as usual 1`] = `
.glamor-0,
[data-glamor-0] {
  font-size: 0.875rem;
  padding-top: 0.0625rem;
  padding-right: 0.1875rem;
  padding-bottom: 0.1875rem;
  padding-left: 0.1875rem;
  background-color: hotpink;
  padding: 10px;
}

<code
  className="glamor-0"
>
  const foo = "bar";
</code>
`;
